x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportMigrateToDataTiersAction.java

```diff
@@ -138,6 +136,8 @@ public void onFailure(Exception e) {
             @Override
             public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
                 super.clusterStateProcessed(oldState, newState);
+                clusterService.getRerouteService().reroute("cluster migrated to data tiers routing", Priority.NORMAL,
+                    ActionListener.noop());
```

Comment:
Does it matter that the listener completes without waiting for the reroute? I expect there to be some tests that need a reroute to have happened - if they're passing today then are they waiting artificially on something else, and if so can the wait be removed?

Reply:
Thanks for bringing this up. 

IMO the API got the job done and shouldn't necessarily wait for a successful reroute - I guess we can have a listener that logs the error? 

Good point about the tests - we tested the effect of the API (ie. some settings got removed, some migrated) - I'll add one that waits for a green cluster after migration. Thanks for the suggestion.